gcc/rust/hir/tree/rust-hir-visitor.cc

```diff
@@ -497,7 +497,7 @@ DefaultHIRVisitor::walk (IfExpr &expr)
 void
 DefaultHIRVisitor::walk (IfExprConseqElse &expr)
 {
-  reinterpret_cast<IfExpr &> (expr).accept_vis (*this);
+  reinterpret_cast<IfExpr &> (expr).IfExpr::accept_vis (*this);
```

Comment:
you don't need the reinterpret_cast here:
```suggestion
  expr.IfExpr::accept_vis (*this);
```